src/parsec.rs

```diff
@@ -160,14 +218,33 @@ impl ParsecMap {
 
     #[cfg(feature = "mock_base")]
     pub fn has_unpolled_observations(&self) -> bool {
-        let parsec = if let Some(parsec) = self.map.values().last() {
+        let parsec = if let Some(parsec) = self.map.values().map(|(p, _)| p).last() {
             parsec
         } else {
             return false;
         };
 
         parsec.has_unpolled_observations()
     }
+
+    fn get_parsec_mut(&mut self, version: u64) -> Option<&mut Parsec> {
+        self.map.get_mut(&version).map(|(p, _)| p)
+    }
+
+    #[allow(unused)]
+    fn needs_pruning(&self, version: u64) -> Option<bool> {
+        self.map.get(&version).map(|(_, s)| s.needs_pruning())
+    }
+
+    #[allow(unused)]
+    fn last_needs_pruning(&self) -> bool {
+        if let Some(s) = self.map.values().last().map(|(_, s)| s.needs_pruning()) {
```

Comment:
do we really need the log_or_panic? or maybe we shall use `unwrap_or` to return `false` in case of empty?

Reply:
Good point, I've changed it to `unwrap_or(false)`